clients/src/main/java/org/apache/kafka/clients/consumer/internals/StreamsGroupHeartbeatRequestManager.java

```diff
@@ -254,34 +284,131 @@ public StreamsGroupHeartbeatRequestManager(final LogContext logContext,
             retryBackoffMaxMs,
             maxPollIntervalMs
         );
+        this.pollTimer = time.timer(maxPollIntervalMs);
     }
 
+    /**
+     * This will build a heartbeat request if one must be sent, determined based on the member
+     * state. A heartbeat is sent in the following situations:
```

Comment:
These are not separate situations in which a heartbeat must be sent. Maybe you mean "when all of the following conditions apply" or soemthign?